lib/providers/auth/app_initialization_provider.dart

```diff
@@ -13,31 +13,59 @@ final retryCounterProvider = StateProvider<int>((ref) => 0);
 
 final appInitializationProvider =
     FutureProvider.family<void, BuildContext>((ref, context) async {
+  var _authProvider = ref.read(authProvider);
   try {
-    var _authProvider = ref.read(authProvider);
+    var retryCount = ref.read(retryCounterProvider);
+    if (retryCount < 2) {
+      _checkInternetConnectivity(context, ref);
+    }
     await _authProvider.initializeUser();
-    var _user = _authProvider.userRes.body as UserTokenModel;
-    _assignNewTokenToDioAndAudioPlayer(ref, _user.token);
-    var deviceInfo = await ref.read(deviceAndAppInfoProvider.future);
-    var data = _setAppOpenedModelData(deviceInfo);
+    var res = _authProvider.userRes;
+    if (res.body != null) {
```

Comment:
When you are setting the userRes to loading in generateUserToken(), the body is null, so won't mean that handleRetry is called here?

`generateUserToken` and `sendOTP` both call `notifyListeners` twice. @osamaasifoak  Is that deliberate?